### ISSUE TITLE
Support for postgres ssl connection

### ISSUE THREAD

**Issue by romankl** [ThomWright/postgres-migrations#25]
Thanks for this awsome project!

Something that creates some problems is the missing SSL support. If your PG instance requires SSL-support, you can't migrate the schema. (Which is often the case in production env's)

This could be somehow solved with the support for connection strings #14  or passing direct ssl options to the underlying pg driver.

**Comment by ThomWright** [comment]
Thanks for the issue. SSL (well, TLS hopefully I guess?) would be good to support. I'd prefer well-typed options, if possible, over connection strings. Are you interested in making a PR for this?

**Comment by austenLacy** [comment]
@romankl @ThomWright technically you can use SSL with the `migrate(...)` function, but not the `createDB(...)` function. Any config you pass to `migrate(...)` actually gets passed directly to the client created by `pg`. See [this line in the migrate function](https://github.com/ThomWright/postgres-migrations/blob/master/src/migrate.ts#L50).

So you can do something like
```javascript
const ca = fs.readFileSync(process.env.SSL_SERVER_CA).toString()
const key = fs.readFileSync(process.env.SSL_CLIENT_KEY).toString()
const cert = fs.readFileSync(process.env.SSL_CLIENT_CERT).toString()

migrate(
  {
    database: process.env.PGDATABASE,
    user: process.env.PGUSER,
    password: process.env.PGPASSWORD,
    host: process.env.PGHOST,
    port: 5432,
    ssl: {
      rejectUnauthorized: false,
      ca: ca,
      key: key,
      cert: cert,
    }
  }, path.resolve(__dirname, './sql')
).then(() => {
  console.log('migration successful')
}).catch((err) => {
  console.log('migration failed: ', err)
})
```

I took a quick glance at `pgtools` which is the library used for the `createDb` and didn't notice any documentation on supporting SSL so it might not unfortunately :( At least this lets you run your migrations though 👍

**Comment by ThomWright** [comment]
Solved by c246ad3 and ad81ed9
